sbids/metrics/c2st.py

```diff
@@ -0,0 +1,128 @@
+
+from tqdm import tqdm
+import numpy as np
+import jax
+import jax.numpy as jnp
+import haiku as hk
+import optax
+from tensorflow_probability.substrates import jax as tfp
+tfd = tfp.distributions
+tfb = tfp.bijectors
+
+class Classifier_C2ST(hk.Module):
+
+  def __call__(self, x):
+
+    net_x = hk.Linear(8)(x) 
+    net_x = jax.nn.leaky_relu(net_x)
+    net_x = hk.Linear(16)(net_x) 
+    net_x = jax.nn.leaky_relu(net_x) 
+    net_x = hk.Linear(32)(net_x) 
+    net_x = jax.nn.leaky_relu(net_x) 
+    net_x = hk.Linear(16)(net_x) 
+    net_x = jax.nn.leaky_relu(net_x) 
+    net_x = hk.Linear(8)(net_x) 
+    net_x = jax.nn.leaky_relu(net_x) 
+    net_x = hk.Linear(1)(net_x) 
+    net_x = jax.nn.sigmoid(net_x).squeeze()
+    net_x = jnp.clip(net_x, 1e-5, 1.-1e-5)
+
+    return net_x
+
+
+def c2st(sample1,sample2,num_seed):
+  """
+  Classifier Two-Sample Test
+
+  Train a binary classifier to evaluate wether sample1 and sample2 
+  are from the same distribution. 
+  Each sample must be of size +-10000 to work properly.
+  Parameters
+   ----------
+   sample1: First sample.
+   sample2 : Second sample.
+   num_seed : seed number
+   Returns
+   -------
+     Accuracy.
+   
+  """
+
+  
+  sample1_mean = jnp.mean(sample1, axis = 0)
+  sample1_std = jnp.std(sample1, axis = 0)
+  sample1 = (sample1 - sample1_mean) / sample1_std
+  sample2 = (sample2 - sample1_mean) / sample1_std
+
+  label1 = jnp.zeros(len(sample1)) 
+  label2 = jnp.ones(len(sample2))  
+  data = jnp.concatenate([jnp.concatenate([sample1,label1.reshape(-1,1)],axis = 1),
+                          jnp.concatenate([sample2,label2.reshape(-1,1)],axis = 1)],axis = 0)
+
+  data = jax.random.permutation(jax.random.PRNGKey(num_seed), data, axis=0)
+  ntrain = int((len(sample1)+len(sample2))*0.6)
+  dtrain, dte = data[:ntrain], data[ntrain:]
+
+  sample_dim = sample1.shape[1]
+  ztrain = dtrain[...,:sample_dim]
+  ltrain = dtrain[...,sample_dim:].squeeze()
```

Comment:
I don't understand why you concatenate the data first, and then split it again. What's the reason for that?

Reply:
you don't need to shuffle at the top. Just use 2 permutations in you `data_stream` function

Reply:
not sure to understand, but it was to create data train and data test

Reply:
oh, I see, yeah because it's after shuffling.

Yeah so my point was more that I wasnt surre concatenating both samples and shuffling them was the best way to do this.  Since you shuffle again below. detail. dtail

Reply:
not very important as long as the 2 saamplees are exactly of the same size